### ISSUE TITLE
Instana blowing up grpc metadata

### ISSUE THREAD

**Issue by joshughes** [instana/nodejs-sensor#132]
We use PubSub and Instana is currently causing issues in our PubSub client by trying to instrument grpc. It seems to inject a ton of metadata and eventually it gets too large and PubSub client can not longer publish. 

```
SEND_INITIAL_METADATA{key=3a 70 61 74 68 ':path' value=2f 67 6f 6f 67 6c 65 2e 70 75 62 73 75 62 2e 76 31 2e 50 75 62 6c 69 73 68 65 72 2f 50 75 62 6c 69 73 68 '/google.pubsub.v1.Publisher/Publish', key=78 2d 67 6f 6f 67 2d 61 70 69 2d 63 6c 69 65 6e 74 'x-goog-api-client' value=67 6c 2d 6e 6f 64 65 2f 76 38 2e 39 2e 34 20 67 72 70 63 2f 31 2e 31 39 2e 30 20 67 61 78 2f 30 2e 32 35 2e 35 20 67 61 70 69 63 2f 30 2e 32 37 2e 30 20 67 63 63 6c 2f 30 2e 32 37 2e 30 'gl-node/v8.9.4 grpc/1.19.0 gax/0.25.5 gapic/0.27.0 gccl/0.27.0', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 61 62 35 38 39 39 37 39 66 62 32 39 31 '000ab589979fb291', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 39 39 37 39 35 33 65 39 34 65 38 37 33 '000997953e94e873', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 66 32 33 62 61 66 61 39 34 36 61 62 62 '000f23bafa946abb', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 35 66 61 30 62 64 32 31 62 31 36 34 66 '0015fa0bd21b164f', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 39 34 31 36 37 39 65 61 32 30 65 66 37 '000941679ea20ef7', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 33 65 30 61 63 66 30 35 65 61 39 37 39 '0003e0acf05ea979', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 30 34 35 39 34 37 35 39 39 30 61 35 39 '0010459475990a59', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 32 32 37 39 38 66 64 66 31 65 61 32 62 '00022798fdf1ea2b', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 35 65 32 37 36 30 64 64 36 64 64 36 66 '0015e2760dd6dd6f', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 38 32 66 39 66 39 30 30 66 31 34 37 37 '00082f9f900f1477', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 32 39 38 34 36 63 38 63 35 63 30 34 39 '00029846c8c5c049', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 30 39 30 30 31 33 66 34 65 62 30 63 35 '000090013f4eb0c5', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 37 66 38 63 37 61 64 65 36 36 62 61 62 '0007f8c7ade66bab', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 37 34 62 62 34 63 65 63 66 35 32 32 31 '00174bb4cecf5221', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 33 35 38 61 36 62 35 35 35 65 65 37 39 '001358a6b555ee79', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 30 39 65 34 62 31 65 31 64 65 64 62 37 '00109e4b1e1dedb7', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 31 34 61 31 33 30 39 65 63 39 36 30 31 '00114a1309ec9601', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 33 34 37 62 31 30 37 61 63 37 30 38 31 '001347b107ac7081', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 38 36 31 31 33 61 66 64 62 63 36 33 37 '00086113afdbc637', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 38 35 34 36 65 33 32 33 61 36 35 31 64 '0018546e323a651d', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 34 34 65 63 35 34 39 63 61 62 31 32 39 '00144ec549cab129', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 34 34 65 63 35 34 39 63 61 62 31 32 39 '00144ec549cab129', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 61 64 36 62 61 34 65 33 39 38 34 34 39 '001ad6ba4e398449', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 61 64 36 62 61 34 65 33 39 38 34 34 39 '001ad6ba4e398449', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 65 34 36 38 37 38 33 34 34 31 61 33 66 '001e468783441a3f', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 65 34 36 38 37 38 33 34 34 31 61 33 66 '001e468783441a3f', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 66 31 38 61 64 35 38 36 31 61 32 37 64 '001f18ad5861a27d', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 66 31 38 61 64 35 38 36 31 61 32 37 64 '001f18ad5861a27d', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 61 39 31 36 30 39 34 33 31 61 38 65 31 '000a91609431a8e1', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 61 39 31 36 30 39 34 33 31 61 38 65 31 '000a91609431a8e1', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 62 61 35 38 63 65 62 37 62 63 33 62 33 '001ba58ceb7bc3b3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 62 61 35 38 63 65 62 37 62 63 33 62 33 '001ba58ceb7bc3b3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 65 64 34 33 66 66 39 34 34 30 34 62 '001ded43ff94404b', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 65 64 34 33 66 66 39 34 34 30 34 62 '001ded43ff94404b', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 66 30 61 65 34 66 33 62 64 39 66 33 '001df0ae4f3bd9f3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 66 30 61 65 34 66 33 62 64 39 66 33 '001df0ae4f3bd9f3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 31 62 63 35 32 38 65 34 39 61 61 39 33 '0011bc528e49aa93', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 31 62 63 35 32 38 65 34 39 61 61 39 33 '0011bc528e49aa93', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 64 39 39 38 33 61 64 33 39 36 30 35 39 '000d9983ad396059', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 64 39 39 38 33 61 64 33 39 36 30 35 39 '000d9983ad396059', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1' deadline=113920}
I0327 16:21:54.334408477     177 channel_stack.cc:226]       OP[client-auth:0x3bbd5a8]:  SEND_INITIAL_METADATA{key=3a 61 75 74 68 6f 72 69 74 79 ':authority' value=70 75 62 73 75 62 2e 67 6f 6f 67 6c 65 61 70 69 73 2e 63 6f 6d 3a 34 34 33 'pubsub.googleapis.com:443', key=3a 70 61 74 68 ':path' value=2f 67 6f 6f 67 6c 65 2e 70 75 62 73 75 62 2e 76 31 2e 50 75 62 6c 69 73 68 65 72 2f 50 75 62 6c 69 73 68 '/google.pubsub.v1.Publisher/Publish', key=78 2d 67 6f 6f 67 2d 61 70 69 2d 63 6c 69 65 6e 74 'x-goog-api-client' value=67 6c 2d 6e 6f 64 65 2f 76 38 2e 39 2e 34 20 67 72 70 63 2f 31 2e 31 39 2e 30 20 67 61 78 2f 30 2e 32 35 2e 35 20 67 61 70 69 63 2f 30 2e 32 37 2e 30 20 67 63 63 6c 2f 30 2e 32 37 2e 30 'gl-node/v8.9.4 grpc/1.19.0 gax/0.25.5 gapic/0.27.0 gccl/0.27.0', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 61 62 35 38 39 39 37 39 66 62 32 39 31 '000ab589979fb291', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 39 39 37 39 35 33 65 39 34 65 38 37 33 '000997953e94e873', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 66 32 33 62 61 66 61 39 34 36 61 62 62 '000f23bafa946abb', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 35 66 61 30 62 64 32 31 62 31 36 34 66 '0015fa0bd21b164f', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 39 34 31 36 37 39 65 61 32 30 65 66 37 '000941679ea20ef7', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 33 65 30 61 63 66 30 35 65 61 39 37 39 '0003e0acf05ea979', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 30 34 35 39 34 37 35 39 39 30 61 35 39 '0010459475990a59', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 32 32 37 39 38 66 64 66 31 65 61 32 62 '00022798fdf1ea2b', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 35 65 32 37 36 30 64 64 36 64 64 36 66 '0015e2760dd6dd6f', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 38 32 66 39 66 39 30 30 66 31 34 37 37 '00082f9f900f1477', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 32 39 38 34 36 63 38 63 35 63 30 34 39 '00029846c8c5c049', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 30 39 30 30 31 33 66 34 65 62 30 63 35 '000090013f4eb0c5', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 37 66 38 63 37 61 64 65 36 36 62 61 62 '0007f8c7ade66bab', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 37 34 62 62 34 63 65 63 66 35 32 32 31 '00174bb4cecf5221', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 33 35 38 61 36 62 35 35 35 65 65 37 39 '001358a6b555ee79', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 30 39 65 34 62 31 65 31 64 65 64 62 37 '00109e4b1e1dedb7', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 31 34 61 31 33 30 39 65 63 39 36 30 31 '00114a1309ec9601', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 33 34 37 62 31 30 37 61 63 37 30 38 31 '001347b107ac7081', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 30 38 36 31 31 33 61 66 64 62 63 36 33 37 '00086113afdbc637', key=78 2d 69 6e 73 74 61 6e 61 2d 73 'x-instana-s' value=30 30 31 38 35 34 36 65 33 32 33 61 36 35 31 64 '0018546e323a651d', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 34 34 65 63 35 34 39 63 61 62 31 32 39 '00144ec549cab129', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 34 34 65 63 35 34 39 63 61 62 31 32 39 '00144ec549cab129', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 61 64 36 62 61 34 65 33 39 38 34 34 39 '001ad6ba4e398449', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 61 64 36 62 61 34 65 33 39 38 34 34 39 '001ad6ba4e398449', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 65 34 36 38 37 38 33 34 34 31 61 33 66 '001e468783441a3f', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 65 34 36 38 37 38 33 34 34 31 61 33 66 '001e468783441a3f', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 66 31 38 61 64 35 38 36 31 61 32 37 64 '001f18ad5861a27d', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 66 31 38 61 64 35 38 36 31 61 32 37 64 '001f18ad5861a27d', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 61 39 31 36 30 39 34 33 31 61 38 65 31 '000a91609431a8e1', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 61 39 31 36 30 39 34 33 31 61 38 65 31 '000a91609431a8e1', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 62 61 35 38 63 65 62 37 62 63 33 62 33 '001ba58ceb7bc3b3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 62 61 35 38 63 65 62 37 62 63 33 62 33 '001ba58ceb7bc3b3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 65 64 34 33 66 66 39 34 34 30 34 62 '001ded43ff94404b', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 65 64 34 33 66 66 39 34 34 30 34 62 '001ded43ff94404b', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 66 30 61 65 34 66 33 62 64 39 66 33 '001df0ae4f3bd9f3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 64 66 30 61 65 34 66 33 62 64 39 66 33 '001df0ae4f3bd9f3', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 31 62 63 35 32 38 65 34 39 61 61 39 33 '0011bc528e49aa93', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 31 31 62 63 35 32 38 65 34 39 61 61 39 33 '0011bc528e49aa93', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 64 39 39 38 33 61 64 33 39 36 30 35 39 '000d9983ad396059', key=78 2d 69 6e 73 74 61 6e 61 2d 74 'x-instana-t' value=30 30 30 64 39 39 38 33 61 64 33 39 36 30 35 39 '000d9983ad396059', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1', key=78 2d 69 6e 73 74 61 6e 61 2d 6c 'x-instana-l' value=31 '1' deadline=113920}
```

**Comment by mgabeler-lee-6rs** [comment]
(I work w/ Joe)

Enabling `GRPC_TRACE` in our app, we can see that with each `gRPC` request, additional `x-instana-t`, `x-instana-s`, and `x-instana-l` headers are being added.  This appears to continue and grow until the headers get so large that the message is too large and Google sends GOAWAY errors to all requests.

For example, from one trace run, I see 126 instana headers logged for one request -- 42 of each of those 3.  Lo and behold, the very next request has 129 instana headres -- 43 of each, naturally.

**Comment by basti1302** [comment]
Thanks for the report. I'll publish a fix for this today.

**Comment by basti1302** [comment]
> with each gRPC request, additional x-instana-t, x-instana-s, and x-instana-l headers are being added

@joshughes @mgabeler-lee-6rs I'm trying to reproduce this locally but failing so far. Can you tell which method exactly you are calling to trigger said gRPC request? (I believe you are using https://www.npmjs.com/package/@google-cloud/pubsub?)

A simplified code snippet of the code snippet would be very helpful.

**Comment by mgabeler-lee-6rs** [comment]
Yes, that is the PubSub client library we are using.  The PubSub calls we make in the application are:

Startup:
```typescript
const pubsub = new PubSub({projectId: /* environment variable */});
const topic = pubsub.topic(/* configuration setting */);

// load named error code map
const protoFiles = require('google-proto-files');
const codesProto = await protoFiles.load(protoFiles('rpc', 'code.proto'));
const googleErrorCodes = codesProto.nested.google.nested.rpc.Code;

// make sure topic exists
await topic.exists();
// real code does something with this result, but in practice the topic is always already setup
```

Runtime:
```typescript
// many things trigger us to do this repeatedly:
const message = {/* 5-20 kb JSON object */};
await topic.publish(Buffer.from(JSON.stringify(message)));
```

**Comment by basti1302** [comment]
A fix for this (which did not require to reproduce the issue) has been published with version 1.63.1. I still want to analyze this a bit more in detail, so thanks for the code snippet.
